extension/agenthealth/handler/stats/provider/statuscode.go

```diff
@@ -0,0 +1,180 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package provider
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/amazon-contributing/opentelemetry-collector-contrib/extension/awsmiddleware"
+
+	"github.com/aws/amazon-cloudwatch-agent/extension/agenthealth/handler/stats/agent"
+)
+
+const (
+	statusResetInterval = 5 * time.Minute
+	statusHandlerID     = "cloudwatchagent.StatusCodeHandler"
+)
+
+var (
+	statusCodeSingleton *StatusCodeHandler
+	statusCodeStatsOnce sync.Once
+)
+
+// StatusCodeHandler provides monitoring for status codes per operation.
+type StatusCodeHandler struct {
+	statsByOperation sync.Map
+	mu               sync.Mutex
+	resetTimer       *time.Timer
+	filter           agent.OperationsFilter
+}
+
+// GetStatusCodeStats retrieves or initializes the singleton StatusCodeHandler.
+func GetStatusCodeStats(filter agent.OperationsFilter) *StatusCodeHandler {
+	statusCodeStatsOnce.Do(func() {
+		handler := &StatusCodeHandler{}
+		handler.filter = filter
+		handler.startResetTimer()
+		statusCodeSingleton = handler
+	})
+	return statusCodeSingleton
+}
+
+// startResetTimer initializes a reset timer to clear stats every 5 minutes.
+func (h *StatusCodeHandler) startResetTimer() {
+	h.resetTimer = time.AfterFunc(statusResetInterval, func() {
+		h.mu.Lock()
+		defer h.mu.Unlock()
+
+		h.statsByOperation.Range(func(key, _ interface{}) bool {
+			h.statsByOperation.Delete(key)
+			return true
+		})
+		log.Println("Status code stats reset.")
+		h.startResetTimer()
```

Comment:
nit - consider using a ticker instead of manually resetting a timer: https://gobyexample.com/tickers